File: src/main/java/com/wikibooks/spark/ch6/SocketSample.java
package com.wikibooks.spark.ch6;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.streaming.Durations;
import org.apache.spark.streaming.api.java.JavaReceiverInputDStream;
import org.apache.spark.streaming.api.java.JavaStreamingContext;

// 6.2.1절 예제 6-5
public class SocketSample {

  public static void main(String[] args) throws Exception {

    SparkConf conf = new SparkConf()
            .setMaster("local[*]")
            .setAppName("SocketSample");

    JavaSparkContext sc = new JavaSparkContext(conf);
    JavaStreamingContext ssc = new JavaStreamingContext(sc, Durations.seconds(3));

    JavaReceiverInputDStream<String> ds =  ssc.socketTextStream("localhost", 9000);

    ds.print();
    
    ssc.start();
    ssc.awaitTermination();
  }
}

File: src/main/scala/com/wikibooks/spark/ch6/scala/SocketSample.scala
package com.wikibooks.spark.ch6.scala

import org.apache.spark.SparkConf
import org.apache.spark.streaming.{Seconds, StreamingContext}

// 6.2.1절 예제 6-4
object SocketSample {

  def main(args: Array[String]): Unit = {

    val conf = new SparkConf().
      setMaster("local[3]")
      .setAppName("SocketSample")

    val ssc = new StreamingContext(conf, Seconds(3))
    val ds = ssc.socketTextStream("localhost", 9000)

    ds.print

    ssc.start()
    ssc.awaitTermination()
  }
}